Add precise swipe actions for swipes from a start to an end point

steviki opened the pull request:
This PR implements and fixes https://github.com/google/EarlGrey/issues/674.

Introduces new swipe actions, that take a `starPoint` and an `endPoint` and perform a swipe along those points via two new public functions:

`grey_swipeFastFromStartToEndPoint(CGPoint startPoint, CGPoint endPoint);`
`grey_swipeSlowFromStartToEndPoint(CGPoint startPoint, CGPoint endPoint);`

---

This is done via the new `GREYPreciseSwipeAction` class, that is implemented similar to `GREYSwipeAction`, but instead of taking a `direction`, `duration` and `startPercents`, the precise swipe action takes a `startPoint`, `endPoint` and `duration`.

When performing this action, a swipe with a single touch is executed that follows the path from the `startPoint` to the `endPoint` in the time of the given `duration`.

steviki commented:
I signed it!

tirodkar commented on EarlGrey/Action/GREYActions.h:
```diff
@@ -178,6 +178,32 @@
+ *  Returns an action that swipes through the view from a given @c startPoint to a given
+ *  @c endPoint.
+ *
+ *  @param startPoint The point where the swipe should begin.
```
Is the startpoint based on a point in the whole screen or within the matched view?

steviki replied:
It is based on the whole screen currently.
But it might make more sense to base this on the origin of the matched view.
I will change that and clarify in the documentation.

tirodkar replied:
@steviki any updates on this?

steviki replied:
@tirodkar Yes, the change to use points based on the origin of the matched view landed in https://github.com/google/EarlGrey/pull/675/commits/0380dbb1785308b82c8b4e33065431629badf515.
So this PR is good to go from my side.

Please let me know if there are any questions left.

tirodkar replied:
QQ - isn't this the same as scroll amount? Ir are you trying to do from any point to another?

steviki commented:
Just wanted to check in on the status here.
I'd love to get this PR merged, so let me know if there is anything left for me to take care of.
Thank you!

khandpur commented:
It looks like you want precision here. Is there a reason for not using scroll API which is much more precise than swipe?

steviki commented:
> Is there a reason for not using scroll API which is much more precise than swipe?

@khandpur I'm not quite sure if I understand your question correctly, so please correct me if I'm wrong. 
If you are referring to `GREYScrollAction`, then the reason is that it doesn't allow scrolling from one point to another in any arbitrary directions, but only in predefined `GREYDirection`s.

khandpur commented on EarlGrey/Action/GREYPreciseSwipeAction.m:
```diff
@@ -0,0 +1,121 @@
+                          duration:(CFTimeInterval)duration {
+
+    NSString *name =
+    [NSString stringWithFormat:@"Precise swipe from %@ to %@ for duration %g",
```
indent 4 spaces

steviki commented:
@khandpur Sorry for the long delay, but I've now added unit and functional tests for the precise swipe actions. 
Looking forward to your feedback!

steviki commented:
@khandpur Just wanted to ping to check on the status of this PR. I would really appreciated if you could have another look at this PR and consider merging it. Let me know if there are any questions!

khandpur commented:
@tirodkar PTAL and merge if it looks good.

khandpur commented:
And thank you Stefan for adding this feature request!

steviki commented:
Thank you for reviewing this PR and getting it merged! I really appreciate it.

@tirodkar I also addressed your review feedback now and fixed the typo.

khandpur commented:
@tirodkar I've restarted the CI. Could we merge this and make it available in earlgrey 2?

steipete commented:
Hi! Are there any blockers left for merging this?
